### PR TITLE
Implement Column Delete and Column Title Edit

### PR DESCRIPTION
**Pull request by tejasj1718**
### What this PR does (required):
- creates the Board UI implemented with several components most importantly Board, Column, Card.
- the columns can be deleted from the column menu
- the column title can be edited.

### Screenshots / Videos (front-end only):

PRBoardUI
PRBoardMenu

PRBoardColumnEdit

PRDeleteCOlumn

### Any information to test information needed to test this feature (required):
- Delete all data beforehand as schema has changed.
- Sign in as any user, demo board data is generated for any user.
- Move the columns and cards around.
- Test the add column buttons by hovering over them on the left and right sides.
- Click on the column menu to either edit or delete the column.
- To reset data, simply delete all columns in the ui and delete the board from the database and restart server.

### Any issues with the current functionality (optional):
- Styling can be improved
- Theming Issues.

### CODE REVIEW

**Review by ezeilo-su — COMMENTED** [review]
This one looks good!

File: client/src/components/Board/Board.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      );
+    }
+  };
+
+  return (
+    <Grid container xs={12} className={classes.root}>
+      <ColumnAddButton location={'left'} />
+      <Grid item xs={10}>
+        <DragDropContext onDragEnd={handleDragEnd}>
+          <Droppable droppableId="column-list" direction="horizontal" type="column">
```
Comment (moffatethan) [inline client/src/components/Board/Board.tsx:44]: Q: Did you write this code or is the base of the branch wrong?

**Review by moffatethan — CHANGES_REQUESTED** [review]
Just the one comment there

File: client/src/components/Board/Board.tsx (hunk 1 above)
Comment (tejasj1718) [reply client/src/components/Board/Board.tsx:44]: I wrote this code, I tried my best to adapt the existing drag and drop feature but ended up rewriting the whole thing. It was influenced by the drag and drop feature that was merged in.

**Review by moffatethan — APPROVED** [review]

**Pull request merged**